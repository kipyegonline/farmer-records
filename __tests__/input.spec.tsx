import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import SignUpForm from "../pages/sign-up";
import UseInput, { SnackBarComponent } from "../utils/input/useInput";
import { TextField } from "@mui/material";

describe("User input", () => {
  beforeEach(() => render(<SignUpForm />));

  it("<Snapshot/>", () => {
    const el = screen.getByTestId("signup-form");
    expect(el).toBeInTheDocument();
    // screen.debug();
  });
});

describe("NaN", () => {
  // expect(false).toBeTruthy();
  it("renders homepage unchanged", () => {
    const { container } = render(<SignUpForm />);
    expect(container).toMatchSnapshot();
  });
});

describe("Snack bar", () => {
  test("Snackbar renders", () => {
    const _close = jest.fn();
    const helper = "Hey No value";
    render(<SnackBarComponent handleClose={_close} helperText={helper} />);
    const textel = screen.getByTestId("snackbar-text");
    const btn = screen.getByRole("button");

    expect(textel).toHaveTextContent(helper);
    fireEvent.click(btn);
    expect(_close).toHaveBeenCalledTimes(1);
    expect(textel).toBeVisible();
  });
});

describe("<TextField/>", () => {
  const _change = jest.fn();
  render(<TextField label="Jest" onChange={_change} value="Jules" />);
  const txf = screen.getByLabelText(/Jest/);
  it("checks value", () => expect(txf).toHaveValue("Vince"));
  fireEvent.change(txf, { value: "Jules" });
  it("changes", () => expect(txf).toHaveValue("Jade"));
});

class MyJules {
  public callMe(val, callback) {
    callback(val);
  }
}

describe("My little class", () => {
  let myclass!: MyJules;
  beforeEach(() => (myclass = new MyJules()));
  it("mocks the gaddam class", () => {
    let _call = jest.fn();
    myclass.callMe("I can make your bed rock", _call);
    expect(_call).toBeCalled();
    expect(_call).toBeCalledWith("My sushi raw");
  });
});
