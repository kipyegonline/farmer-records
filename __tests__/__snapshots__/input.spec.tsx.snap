// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NaN renders homepage unchanged 1`] = `
<div>
  <div
    class="p-4 mt-8 flex flex-col justify-evenly flex-1  md:flex-row-reverse md:mt-2 MuiBox-root css-0"
  >
    <div
      class="my-2 MuiBox-root css-0"
    >
      <form
        class="flex flex-col justify-evenly items-start p-4  bg-white"
        data-testid="notes-form"
        style="max-width: 400px;"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph text-center css-z2eky3-MuiTypography-root"
        >
          Add Notes on farm progress today
        </p>
        <div
          class="p-2 m-2"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 text-center css-ag7rrr-MuiTypography-root"
          >
            Add Farm activity Activity
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Presence of pests and diseases?
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Fruits? Flowers? weeds? 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Heavy rainfall or drought?
          </p>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root my-2 py-1 css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":rf:"
              name="date"
              type="date"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root my-2 py-1 css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              id=":rg:"
              name="Note"
              rows="3"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root bg-blue-600 w-full css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Add Note
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div />
      </form>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <form
        class="bg-white p-8 mx-auto flex flex-col justify-evenly "
        style="max-width: 400px; padding: 2rem;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root my-2  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rh:"
            id=":rh:-label"
          >
            Enter activity name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":rh:"
              name="activity"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Enter activity name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root my-2  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":ri:"
            id=":ri:-label"
          >
            date
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":ri:"
              name="date"
              type="date"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1in441m"
              >
                <span>
                  date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root my-2  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":rj:"
            id=":rj:-label"
          >
            Enter cost expense
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
              >
                Ksh.
              </p>
            </div>
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
              id=":rj:"
              name="totalCost"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1in441m"
              >
                <span>
                  Enter cost expense
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root my-2  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rk:"
            id=":rk:-label"
          >
            Details
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              id=":rk:"
              name="description"
              rows="3"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Details
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root bg-blue-600 mt-4 mb-2 css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Add Activity
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="my-2"
        />
      </form>
    </div>
    <div
      class="my-2 mx-auto MuiBox-root css-0"
    >
      <form
        class="flex flex-col flex-1  justify-evenly bg-white p-4"
        data-testid="input-form"
        style="max-width: 400px;"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 text-center css-10wpov9-MuiTypography-root"
        >
          Add project farm inputs
        </h6>
        <div
          class="MuiFormControl-root MuiTextField-root my-2 py-1  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":rl:"
            id=":rl:-label"
          >
            Enter input name
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-describedby=":rl:-helper-text"
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              id=":rl:"
              name="input"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id=":rl:-helper-text"
          >
            You can use commas to separate your items.
          </p>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root my-2 py-1  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":rm:"
            id=":rm:-label"
          >
            date
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-describedby=":rm:-helper-text"
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              id=":rm:"
              name="date"
              type="date"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id=":rm:-helper-text"
          >
            Add date of input purchase
          </p>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root my-2 mt-4 py-1  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":rn:"
            id=":rn:-label"
          >
            cost of input
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1xab92c-MuiInputBase-root-MuiFilledInput-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium font-bold css-9mc7sf-MuiInputAdornment-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
              >
                Ksh
              </p>
            </div>
            <input
              aria-describedby=":rn:-helper-text"
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
              id=":rn:"
              name="cost"
              type="number"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
            id=":rn:-helper-text"
          >
            Add date of input purchase
          </p>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root bg-blue-600 css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Add input
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div />
      </form>
    </div>
    <form
      class="bg-white p-2 flex justify-evenly items-stretch  mx-auto mt-15 flex-1 flex-col border-2 border-white"
      data-testid="signup-form"
      style="max-width: 400px;"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      />
      <h3
        class="MuiTypography-root MuiTypography-h3 text-center p-0 m-2 css-gepadz-MuiTypography-root"
      >
        Join form
      </h3>
      <div
        class="MuiFormControl-root MuiTextField-root my-1 py-1 px-1 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="name"
          id="name-label"
        >
          Enter name
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            id="name"
            name="Name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root my-1 py-1 px-1 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="username"
          id="username-label"
        >
          Enter username
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            id="username"
            name="User name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root my-1 py-1 px-1 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="email"
          id="email-label"
        >
          Enter email
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            id="email"
            name="Email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root my-1 py-1 px-1 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="phone"
          id="phone-label"
        >
          Enter mobile phone number
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            id="phone"
            name="Phone number"
            type="telephone"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root my-1 py-1 px-1 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Enter password
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            id="password"
            name="Password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root my-1 py-1 px-1 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="cpassword"
          id="cpassword-label"
        >
          Enter password
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-19mk8g1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            id="cpassword"
            name="Confirmation password"
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root my-2 mt-6 py-2 px-1 bg-blue-700  css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Submit
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="flex justify-center"
      >
        <a
          href="/"
        >
          Login
        </a>
      </div>
    </form>
  </div>
</div>
`;
